### DISCUSSION TITLE
Can this package be used outside of the sanity repo

### DISCUSSION THREAD

**Discussion by aidenBarrett96** [ricokahler/sanity-codegen#186] [Q&A] [answered]
I am creating a repo to easily create sanity instances with our component library, I would like to extend the functionality of `sanity-codegen` from our context. So we can generate the types from our repo.

**Comment by ricokahler** [comment, marked as answer]
yes it can! it doesn't have any dependencies on sanity actually. it's a just normal NPM package you can install anywhere

you do need to have your schema files available for code generator to read but if you wanted to lift those js schema files to some common repo (e.g. component lib repo) and import them in the studio code, that would result in the same codegen.

alternatively, you could codegen types from your studio repo and then publish some sort of package that you could import in your component library repo.

let me know if that makes sense
